sphinx/search/__init__.py

```diff
@@ -504,7 +504,7 @@ def _visit_nodes(node):
             elif isinstance(node, nodes.title):
                 title = node.astext()
                 ids = node.parent['ids']
-                word_store.titles.append((title, ids[0] if ids else None))
+                word_store.titles.append((title, ids[0] if ids and word_store.titles else None))  # NoQA: E501
```

Comment:
Just split the line into to (in general, I only use E501 for signatures, not for calls like that).